src/dispatcher.ts

```diff
@@ -216,23 +219,30 @@ export class Dispatcher {
   }
 
   async _obtainWorker() {
-    // If there is worker, use it.
-    if (this._freeWorkers.length)
-      return this._freeWorkers.pop();
-    // If we can create worker, create it.
-    if (this._workers.size < this._config.workers)
-      this._createWorker();
-    // Wait for the next available worker.
-    await new Promise(f => this._workerClaimers.push(f));
-    return this._freeWorkers.pop();
+    const claimWorker = async () => {
+      // Use available worker.
+      if (this._freeWorkers.length)
+        return this._freeWorkers.pop();
+      // Create a new worker.
+      if (this._workers.size < this._config.workers)
+        return this._createWorker();
+      return null;
+    };
+
+    let worker = await claimWorker();
```

Comment:
Drop await, otherwise you could get _notifyWorkerClaimer called while you are in a microtask.